### PR TITLE
Organize jobs in e2e.sh

### PR DESCRIPTION
**Pull request by ikehz**
First part of #20353, isolated from reworking the skip lists.

Work toward #10548.

### CODE REVIEW

**Comment by ihmccreery** [comment]
@ixdy I'd like to get this in ASAP to avoid dangerous rebases.

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -317,48 +341,28 @@ case ${JOB_NAME} in
...
-    : ${E2E_CLUSTER_NAME:="jenkins-gce-e2e-examples"}
-    : ${E2E_NETWORK:="e2e-examples"}
-    : ${GINKGO_TEST_ARGS:="--ginkgo.focus=\[Feature:Example\]"}
-    : ${KUBE_GCE_INSTANCE_PREFIX:="e2e-examples"}
-    : ${PROJECT:="kubernetes-jenkins"}
+  # Runs slow tests on GCE, sequentially.
+  kubernetes-e2e-gce-slow)
+    : ${E2E_CLUSTER_NAME:="jenkins-gce-e2e-slow"}
+    : ${E2E_NETWORK:="e2e-slow"}
+    # TODO(ihmccreery) remove [Skipped] once tetss are relabeled
```
Comment (spxtr) [inline hack/jenkins/e2e.sh]: typo

**Comment by spxtr** [comment]
Looks fine, but I'm worried about that e2e run.

@k8s-bot e2e test this again #IGNORE

**Comment by ihmccreery** [comment]
Ping @spxtr

**Comment by ixdy** [comment]
LGTM

**Comment by ixdy** [comment]
(the typo was already there. can fix in follow-up PRs)

**Comment by spxtr** [comment]
LGTM

**Comment by ihmccreery** [comment]
@k8s-oncall please consider merging to avoid need to rebase.

**Pull request merged**